staging/vhost-device-video/src/vhu_video.rs

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use crate::vhu_video_thread::VhostUserVideoThread;
+use crate::video_backends;
+use clap::ValueEnum;
+use log::{debug, warn};
+use std::{
+    convert,
+    io::{self, Result as IoResult},
+    path::Path,
+    sync::{Arc, Mutex, RwLock},
+};
+use thiserror::Error as ThisError;
+use vhost::vhost_user::message::{VhostUserProtocolFeatures, VhostUserVirtioFeatures};
+use vhost_user_backend::{VhostUserBackendMut, VringRwLock, VringT};
+use virtio_bindings::bindings::virtio_config::{VIRTIO_F_NOTIFY_ON_EMPTY, VIRTIO_F_VERSION_1};
+use virtio_bindings::bindings::virtio_ring::{
+    VIRTIO_RING_F_EVENT_IDX, VIRTIO_RING_F_INDIRECT_DESC,
+};
+use virtio_queue::DescriptorChain;
+use vm_memory::{ByteValued, GuestMemoryAtomic, GuestMemoryLoadGuard, GuestMemoryMmap, Le32};
+use vmm_sys_util::epoll::EventSet;
+use vmm_sys_util::eventfd::{EventFd, EFD_NONBLOCK};
```

Comment:
Using `rustfmt` with nightly (e.g. `cargo +nightly fmt`) with this [`rustfmt.toml`](https://github.com/rust-vmm/vhost-device/blob/main/staging/vhost-device-sound/rustfmt.toml) should fix these up for you.

Reply:
@epilys maybe we can add that in the custom-pipeline we just merged (for both main and staging workspace). IIUC the container has also the nightly toolchain

Reply:
that will be really helpful and save us some time

Reply:
@vireshk yeah!
So we can move the https://github.com/rust-vmm/vhost-device/blob/main/staging/vhost-device-sound/rustfmt.toml in the root and add a new step in the custom-pipeline.
@epilys do you want to take a look? Otherwise, I can do it.

Reply:
Note that I have also added the file for the video crate, following @epilys suggestion. We can remove the video copy too with the proposed update to the CIs.

Reply:
@aesteve-rh thanks for letting me know, I'll open an issue to not forget ;-)

Reply:
> @vireshk yeah! So we can move the https://github.com/rust-vmm/vhost-device/blob/main/staging/vhost-device-sound/rustfmt.toml in the root and add a new step in the custom-pipeline. @epilys do you want to take a look? Otherwise, I can do it.

We'd need to add a step that checks there's no difference between what is committed and what rustfmt outputs; and according to the documentation moving the file to a parent directory should work. It doesn't say anything about workspaces so we should check that.

> Rustfmt is designed to be very configurable. You can create a TOML file called rustfmt.toml or .rustfmt.toml, place it in the project or any other parent directory and it will apply the options in that file. [^0]

I can pick it up, I will update the issue if/when I do.

[^0]: https://rust-lang.github.io/rustfmt/?version=v1.4.36&search=

Reply:
@epilys thanks! I'll assign to you #488